### PR TITLE
modules: update cmsis-nn, cmsis-dsp, and tflite-micro

### PR DESCRIPTION
**Pull request by XenuIsWatching**
Related Issue: https://github.com/zephyrproject-rtos/zephyr/issues/56917

CMSIS-NN and CMSIS-DSP have been moved out of CMSIS. This brings them up to date with the latest of those repos.
A dependency of updating cmsis-nn also required tflite-micro to be updated.

New Zephyr Modules have already been created for cmsis-dsp and cmsis-nn found here:
https://github.com/zephyrproject-rtos/cmsis-nn/pull/1
https://github.com/zephyrproject-rtos/cmsis-dsp/pull/1

cmsis also had to be updated to remove the cmsis-nn and cmsis-dsp code within it
https://github.com/zephyrproject-rtos/cmsis/pull/21

tflite-micro has also been updated to the latest head (at the time of the pull) here:
https://github.com/zephyrproject-rtos/tflite-micro/pull/4

This did require updating test cases for tflite-micro as some apis were deprecated or 'moved around'

cmsis-nn and cmsis-dsp now use the cmake-ext with the cmake defined within the zephyr repo

### CODE REVIEW

**Comment by XenuIsWatching** [comment]
wow.... ok, well. I may have uncovered a likely compiler bug. I'm quite sure how to proceed from here
```
ccache /opt/toolchains/zephyr-sdk-0.16.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DARM_MATH_LOOPUNROLL -DKERNEL -DTC_RUNID=90f4392f064e00bf4181a16b857c89d0 -D_ANSI_SOURCE -D_FORTIFY_SOURCE=1 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -I/__w/zephyr/modules/lib/cmsis-dsp/Source/DistanceFunctions -I/__w/zephyr/modules/lib/cmsis-dsp/Source/../Include -I/__w/zephyr/modules/hal/cmsis/CMSIS/Core/Include -I/__w/zephyr/modules/lib/cmsis-dsp/Source/../PrivateInclude -I/__w/zephyr/zephyr/include -I/__w/zephyr/zephyr/twister-out/mps3_an547/tests/subsys/dsp/basicmath/zdsp.basicmath.fpu/zephyr/include/generated -I/__w/zephyr/zephyr/soc/arm/arm/mps3 -I/__w/zephyr/zephyr/lib/libc/newlib/include -I/__w/zephyr/zephyr/subsys/testsuite/include -I/__w/zephyr/zephyr/subsys/testsuite/ztest/include -I/__w/zephyr/zephyr/subsys/dsp/cmsis/public -I/__w/zephyr/modules/lib/cmsis-dsp/Include -I/__w/zephyr/modules/lib/cmsis-dsp/PrivateInclude -fno-strict-aliasing -Werror -Os -imacros /__w/zephyr/zephyr/twister-out/mps3_an547/tests/subsys/dsp/basicmath/zdsp.basicmath.fpu/zephyr/include/generated/autoconf.h -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m55 -mthumb -mabi=aapcs -mfpu=auto -mfloat-abi=hard -mfp16-format=ieee --sysroot=/opt/toolchains/zephyr-sdk-0.16.0/arm-zephyr-eabi/arm-zephyr-eabi -imacros /__w/zephyr/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/__w/zephyr/zephyr/tests/subsys/dsp/basicmath=CMAKE_SOURCE_DIR -fmacro-prefix-map=/__w/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/__w/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections -mcmse -std=c99 -flax-vector-conversions -MD -MT modules/cmsis-dsp/CMakeFiles/..__modules__lib__cmsis-dsp.dir/Source/DistanceFunctions/arm_dtw_distance_f32.c.obj -MF modules/cmsis-dsp/CMakeFiles/..__modules__lib__cmsis-dsp.dir/Source/DistanceFunctions/arm_dtw_distance_f32.c.obj.d -o modules/cmsis-dsp/CMakeFiles/..__modules__lib__cmsis-dsp.dir/Source/DistanceFunctions/arm_dtw_distance_f32.c.obj -c /__w/zephyr/modules/lib/cmsis-dsp/Source/DistanceFunctions/arm_dtw_distance_f32.c
/tmp/ccwUTx6v.s: Assembler messages:
/tmp/ccwUTx6v.s:172: Error: syntax error -- `vcvtne.f32.s32 s15,s15'
```

**Comment by stephanosio** [comment]
> wow.... ok, well... I may have uncovered a likely compiler bug. I noticed this on my local machine when running it, but it shows up here on twister build (20) here.

It looks you did. Can you please file a bug to the [sdk-ng repository](https://github.com/zephyrproject-rtos/sdk-ng)?

p.s. It might be related to https://bugs.launchpad.net/gcc-arm-embedded/+bug/1982289 (try without `-Os`).

**Comment by XenuIsWatching** [comment]
> p.s. It might be related to https://bugs.launchpad.net/gcc-arm-embedded/+bug/1982289 (try without `-Os`)

I did find that link, but there was no update on the status from ARM on that issue other than "We are able to reproduce the mentioned issue and are working on fixing it." back in August. Nice find with the issue with gas already being known.

This should be compiling with `-Ofast` according to the cmsis-dsp readme as I thought I added as a compile option in the cmake, but I do see that `-Os` coming in there. I'd have to investigate where that is coming from in the build_system

**Comment by stephanosio** [comment]
> > wow.... ok, well... I may have uncovered a likely compiler bug. I noticed this on my local machine when running it, but it shows up here on twister build (20) here.
> 
> ~It looks you did. Can you please file a bug to the [sdk-ng repository](https://github.com/zephyrproject-rtos/sdk-ng)?~ Filed [zephyrproject-rtos/sdk-ng#665](https://github.com/zephyrproject-rtos/sdk-ng/issues/665) to track this.
> 
> p.s. It might be related to https://bugs.launchpad.net/gcc-arm-embedded/+bug/1982289 (try without `-Os`). p.s. It looks like it is a bug in gas, which has been fixed in [bminor/binutils-gdb@656412a](https://github.com/bminor/binutils-gdb/commit/656412a7a4da0aef43ead1ea976ba0235a7ec30a)

https://github.com/zephyrproject-rtos/sdk-ng/pull/666

**Comment by XenuIsWatching** [comment]
So, I realized I put the -Ofast in the CMSIS-NN as recommended, but forgot CMSIS-DSP. adding `zephyr_library_compile_options(-Ofast)` in to the cmake for cmsis-dsp just gets around the issue for now

**Comment by XenuIsWatching** [comment]
Well... this is a game of a whack-a-compiler-bug.

After, enabling `-Ofast` on cmsis-dsp, now there is a internal compiler error
```
/__w/zephyr/modules/lib/cmsis-dsp/Source/MatrixFunctions/arm_mat_mult_f64.c: In function 'arm_mat_mult_f64':
/__w/zephyr/modules/lib/cmsis-dsp/Source/MatrixFunctions/arm_mat_mult_f64.c:506:1: internal compiler error: in arc_split_move, at config/arc/arc.cc:10235
  506 | }
      | ^
0x7ff86c016d8f __libc_start_call_main
	../sysdeps/nptl/libc_start_call_main.h:58
0x7ff86c016e3f __libc_start_main_impl
	../csu/libc-start.c:392
Please submit a full bug report, with preprocessed source (by using -freport-bug).
Please include the complete backtrace with any bug report.
See <https://github.com/zephyrproject-rtos/sdk-ng/issues> for instructions.
```

**Comment by XenuIsWatching** [comment]
it appears cmsis-dsp is getting dinged, but those issues were fixed a couple of years ago https://github.com/ARM-software/CMSIS-DSP/commit/5fac45cc967f2f6b8592856a32ce04fae2245143

these should all be fixed within zephyr with the https://github.com/zephyrproject-rtos/zephyr/pull/56957

**Comment by XenuIsWatching** [comment]
I found out `ARM_MATH_LOOPUNROLL` was getting enabled by default....

I removed the default enable and that seems to workaround the compiler bug :shrug:

**Comment by stephanosio** [comment]
The CMSIS-DSP and CMSIS-NN Zephyr module repositories have been created as per https://github.com/zephyrproject-rtos/zephyr/issues/50928 and https://github.com/zephyrproject-rtos/zephyr/issues/56917, respectively:

* https://github.com/zephyrproject-rtos/cmsis-dsp/
* https://github.com/zephyrproject-rtos/cmsis-nn/

**Comment by stephanosio** [comment]
> The CMSIS-DSP and CMSIS-NN Zephyr module repositories have been created as per #50928 and #56917, respectively:
> 
> * https://github.com/zephyrproject-rtos/cmsis-dsp/
> * https://github.com/zephyrproject-rtos/cmsis-nn/

@XenuIsWatching Can you please create pull requests to the `zephyr` branch of these module repositories?

File: modules/Kconfig.cmsis_dsp
Comment (stephanosio) [inline modules/Kconfig.cmsis_dsp:60]: What is the rationale for removing these configs?

File: modules/Kconfig.cmsis_dsp
Comment (stephanosio) [inline modules/Kconfig.cmsis_dsp:21]: Why do we have both `CMSIS_DSP_NEON` and `CMSIS_DSP_NEON_EXPERIMENTAL`?

**Review by stephanosio — DISMISSED** [review]

File: modules/Kconfig.cmsis_dsp
Comment (XenuIsWatching) [reply modules/Kconfig.cmsis_dsp:21]: In case if someone wants to enable the neon experimental....? This is an option defined within CMSIS-DSP

**Comment by XenuIsWatching** [comment]
> > The CMSIS-DSP and CMSIS-NN Zephyr module repositories have been created as per #50928 and #56917, respectively:
> > 
> > * https://github.com/zephyrproject-rtos/cmsis-dsp/
> > * https://github.com/zephyrproject-rtos/cmsis-nn/
> 
> @XenuIsWatching Can you please create pull requests to the `zephyr` branch of these module repositories?
> 
> p.s. same goes for `cmsis` and `tflite-micro`

Done, PRs are as follows:

https://github.com/zephyrproject-rtos/cmsis/pull/21
https://github.com/zephyrproject-rtos/tflite-micro/pull/4
https://github.com/zephyrproject-rtos/cmsis-nn/pull/1
https://github.com/zephyrproject-rtos/cmsis-dsp/pull/1

**Comment by XenuIsWatching** [comment]
What are our options here? Our current blocker is @laurenmurphyx64 not being responsive for https://github.com/zephyrproject-rtos/tflite-micro/pull/4

**Comment by JordanYates** [comment]
> What are our options here? Our current blocker is @laurenmurphyx64 not being responsive for [zephyrproject-rtos/tflite-micro#4](https://github.com/zephyrproject-rtos/tflite-micro/pull/4)

The current blocker is the `west.yml` file. You need to update this to point back to the `zephyrproject` repositories, with the revision changed to the PR like so:
```
    - name: hal_nordic
      revision: pull/137/head
      path: modules/hal/nordic
```
Once CI is passing with `west.yml` updated, those PR's can be properly merged.
Once the PR's are merged, there is a final update to point at the actual commit hash, instead of the PR head.

**Comment by XenuIsWatching** [comment]
> > What are our options here? Our current blocker is @laurenmurphyx64 not being responsive for [zephyrproject-rtos/tflite-micro#4](https://github.com/zephyrproject-rtos/tflite-micro/pull/4)
> 
> The current blocker is the `west.yml` file. You need to update this to point back to the `zephyrproject` repositories, with the revision changed to the PR like so:
> 
> ```
>     - name: hal_nordic
>       revision: pull/137/head
>       path: modules/hal/nordic
> ```
> 
> Once CI is passing with `west.yml` updated, those PR's can be properly merged. Once the PR's are merged, there is a final update to point at the actual commit hash, instead of the PR head.

yes........ but in order for that to happen these need to be merged in first

https://github.com/zephyrproject-rtos/cmsis/pull/21
https://github.com/zephyrproject-rtos/cmsis-nn/pull/1
https://github.com/zephyrproject-rtos/cmsis-dsp/pull/1
https://github.com/zephyrproject-rtos/tflite-micro/pull/4,

**Comment by XenuIsWatching** [comment]
> > What are our options here? Our current blocker is @laurenmurphyx64 not being responsive for [zephyrproject-rtos/tflite-micro#4](https://github.com/zephyrproject-rtos/tflite-micro/pull/4)
> 
> The current blocker is the `west.yml` file. You need to update this to point back to the `zephyrproject` repositories, with the revision changed to the PR like so:
> 
> ```
>     - name: hal_nordic
>       revision: pull/137/head
>       path: modules/hal/nordic
> ```
> 
> Once CI is passing with `west.yml` updated, those PR's can be properly merged. Once the PR's are merged, there is a final update to point at the actual commit hash, instead of the PR head.

I'll update it... but they are pointing to the same hash anyway so it's the same test anyway..

**Comment by XenuIsWatching** [comment]
Samples have been updated as now `AllOpsResolver` is now deprecated and removed in the latest tflite-micro. The samples are now using the `MicroMutableOpResolver` with their respective kernels that they need.

IMO this is actually a better sample as the `AllOpsResolver` will bloat the code by ALOT ( > 200kB) and should only be used while the model is still in development. `MicroMutableOpResolver` is much more efficient by that you have to specific which kernels are used (which can be viewed with a tool such as Netron on the *.tflite file) and the linker will only collect those kernels.

**Comment by cfriedt** [comment]
Added the `TSC` label to hopefully push https://github.com/zephyrproject-rtos/tflite-micro/pull/4 along

**Comment by JordanYates** [comment]
General comment for `cmsis-nn` and `cmsis-dsp`, I think it makes more sense to leave the build systems in those repos as they are and instead do all the CMake configuration in the main repo.
```
name: cmsis-nn
build:
  cmake-ext: True
  kconfig-ext: True
```

**Comment by XenuIsWatching** [comment]
> General comment for `cmsis-nn` and `cmsis-dsp`, I think it makes more sense to leave the build systems in those repos as they are (but unused) and instead do all the CMake configuration in the main repo.
> 
> ```
> name: cmsis-nn
> build:
>   cmake-ext: True
>   kconfig-ext: True
> ```

This was something I did consider, and it does make more sense to me as well. I did that way because that's what the `cmsis` module was doing before. 
I'll look in to when I get time.

**Comment by nashif** [comment]
@cfriedt  what is the ask for the TSC?

**Comment by cfriedt** [comment]
> @cfriedt  what is the ask for the TSC?

Oh, @XenuIsWatching has been waiting on review in the tflite repo for 2 months. I was hoping we could bring it up to see if there is anyone available for review. We assumed @laurenmurphyx64 was not available, but is there someone else who is?

**Comment by cfriedt** [comment]
> General comment for `cmsis-nn` and `cmsis-dsp`, I think it makes more sense to leave the build systems in those repos as they are (but unused) and instead do all the CMake configuration in the main repo.
> ```
> name: cmsis-nn
> build:
>   cmake-ext: True
>   kconfig-ext: True
> ```

I can vouch for this approach - it's significantly easier and allows you to do some minor "patching" using Zephyr-specific sources.

**Comment by cfriedt** [comment]
> @cfriedt  what is the ask for the TSC?

@nashif - @XenuIsWatching has been waiting on review in the tflite repo for 2 months. I was hoping we could bring it up to see if there is anyone available for review. @laurenmurphyx64 was not available, but is there someone else who is?

File: modules/Kconfig.cmsis_dsp
Comment (cfriedt) [reply modules/Kconfig.cmsis_dsp:21]: If there is a Kconfig option for it, then it should probably be OK. I guess some additional cflags are necessary to enable the relevant builtins? I guess that is potentially @stephanosio's main concern - do we need additional toolchain profiles in the SDK?

**Comment by XenuIsWatching** [comment]
> General comment for `cmsis-nn` and `cmsis-dsp`, I think it makes more sense to leave the build systems in those repos as they are (but unused) and instead do all the CMake configuration in the main repo.
> 
> ```
> name: cmsis-nn
> build:
>   cmake-ext: True
>   kconfig-ext: True
> ```

Done for `cmsis-nn`, I still need to do `cmsis-dsp` when I can get around to it

File: modules/cmsis_dsp/CMakeLists.txt
Comment (JordanYates) [inline modules/cmsis_dsp/CMakeLists.txt:56]: Is there a reason you are manually listing all the files instead of just globbing everything in `${CMSIS_DSP_DIR}/Source/BasicMathFunctions`?

Comment (XenuIsWatching) [reply modules/cmsis_dsp/CMakeLists.txt:56]: I could :shrug: ... but that is just how the upstream does it

**Review by cfriedt — APPROVED** [review]

**Comment by XenuIsWatching** [comment]
Why does the zephyr bot keep adding the the "DMN" tag?

**Review by JordanYates — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Comment by cfriedt** [comment]
> Why does the zephyr bot keep adding the the "DNM" tag?

It's a problem with the mobile app caching state. Check the GH website and it's green 👍

**Comment by cfriedt** [comment]
> Why does the zephyr bot keep adding the the "DNM" tag?

The mobile app actually caches state incorrectly. Currently, the app shows 1 check failing, but all tests are clearly passing on the website.

**Comment by cfriedt** [comment]
Resisting the urge to immediately merge this. It's been cooking a while and has had plenty of time for review but the new policy says someone from a different org should merge.

**Comment by XenuIsWatching** [comment]
My only concern is with the branch naming of tflite-micro that happened with the merge here https://github.com/zephyrproject-rtos/tflite-micro/pull/4

It looks like it got merged in to `main` rather than a new `zephyr` branch.... but since the `west.yml` points at a hash. I don't think it matters to much. I don't have the power to force branch points to move around :(

**Review by carlescufi — APPROVED** [review]

**Comment by kartben** [comment]
@XenuIsWatching 
I am confused as to why this is not caught by CI, but I think this is breaking docs build for me locally, with mentions to something `ZEPHYR_CMSIS_DSP_KCONFIG` something (see below)

Shouldn't `modules/Kconfig` have been updated to add a `source "modules/cmsis_dsp/Kconfig"` entry? (note that adding it does not seem to be quite enough to fix things)

```
ninja html
[0/2] Generating Devicetree bindings documentation...
[1/2] Running Sphinx HTML build...Running Sphinx v6.2.1
Building Kconfig database...... failed

Traceback (most recent call last):
  File "C:\Users\kartben\zephyrproject\zephyr\scripts\kconfig\kconfiglib.py", line 2188, in _enter_file
    self._readline = self._open(filename, "r").readline
                     ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kartben\zephyrproject\zephyr\scripts\kconfig\kconfiglib.py", line 3921, in _open
    open(filename, mode, encoding=self._encoding)
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\kartben\\zephyrproject\\zephyr\\'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\kartben\zephyrproject\zephyr\doc\_extensions\zephyr\kconfig\__init__.py", line 117, in kconfig_load
    return kconfiglib.Kconfig(ZEPHYR_BASE / "Kconfig"), module_paths
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kartben\zephyrproject\zephyr\scripts\kconfig\kconfiglib.py", line 947, in __init__
    self._init(filename, warn, warn_to_stderr, encoding)
  File "C:\Users\kartben\zephyrproject\zephyr\scripts\kconfig\kconfiglib.py", line 1085, in _init
    self._parse_block(None, self.top_node, self.top_node).next = None
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kartben\zephyrproject\zephyr\scripts\kconfig\kconfiglib.py", line 2988, in _parse_block
    prev = self._parse_block(None, parent, prev)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kartben\zephyrproject\zephyr\scripts\kconfig\kconfiglib.py", line 3027, in _parse_block
    self._parse_block(_T_ENDMENU, node, node)
  File "C:\Users\kartben\zephyrproject\zephyr\scripts\kconfig\kconfiglib.py", line 2988, in _parse_block
    prev = self._parse_block(None, parent, prev)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kartben\zephyrproject\zephyr\scripts\kconfig\kconfiglib.py", line 2988, in _parse_block
    prev = self._parse_block(None, parent, prev)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kartben\zephyrproject\zephyr\scripts\kconfig\kconfiglib.py", line 3027, in _parse_block
    self._parse_block(_T_ENDMENU, node, node)
  File "C:\Users\kartben\zephyrproject\zephyr\scripts\kconfig\kconfiglib.py", line 2987, in _parse_block
    self._enter_file(filename)
  File "C:\Users\kartben\zephyrproject\zephyr\scripts\kconfig\kconfiglib.py", line 2191, in _enter_file
    raise _KconfigIOError(
kconfiglib._KconfigIOError: C:\Users\kartben\AppData\Local\Temp\tmp96ltv01q/Kconfig.modules:26: Could not open 'C:\Users\kartben\zephyrproject\zephyr\' (in 'osource "$(ZEPHYR_CMSIS_DSP_KCONFIG)"') (ENOENT: No such file or directory)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python311\Lib\shutil.py", line 620, in _rmtree_unsafe
    os.unlink(fullname)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\kartben\\AppData\\Local\\Temp\\tmp96ltv01q\\Kconfig.modules'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python311\Lib\tempfile.py", line 878, in onerror
    _os.unlink(path)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\kartben\\AppData\\Local\\Temp\\tmp96ltv01q\\Kconfig.modules'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\kartben\zephyrproject\.venv\Lib\site-packages\sphinx\events.py", line 96, in emit
    results.append(listener.handler(self.app, *args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kartben\zephyrproject\zephyr\doc\_extensions\zephyr\kconfig\__init__.py", line 247, in kconfig_build_resources
    kconfig, module_paths = kconfig_load(app)
                            ^^^^^^^^^^^^^^^^^
  File "C:\Users\kartben\zephyrproject\zephyr\doc\_extensions\zephyr\kconfig\__init__.py", line 69, in kconfig_load
    with TemporaryDirectory() as td:
  File "C:\Python311\Lib\tempfile.py", line 904, in __exit__
    self.cleanup()
  File "C:\Python311\Lib\tempfile.py", line 908, in cleanup
    self._rmtree(self.name, ignore_errors=self._ignore_cleanup_errors)
  File "C:\Python311\Lib\tempfile.py", line 890, in _rmtree
    _shutil.rmtree(name, onerror=onerror)
  File "C:\Python311\Lib\shutil.py", line 759, in rmtree
    return _rmtree_unsafe(path, onerror)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\shutil.py", line 622, in _rmtree_unsafe
    onerror(os.unlink, fullname, sys.exc_info())
  File "C:\Python311\Lib\tempfile.py", line 881, in onerror
    cls._rmtree(path, ignore_errors=ignore_errors)
  File "C:\Python311\Lib\tempfile.py", line 890, in _rmtree
    _shutil.rmtree(name, onerror=onerror)
  File "C:\Python311\Lib\shutil.py", line 759, in rmtree
    return _rmtree_unsafe(path, onerror)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\shutil.py", line 603, in _rmtree_unsafe
    onerror(os.scandir, path, sys.exc_info())
  File "C:\Python311\Lib\shutil.py", line 600, in _rmtree_unsafe
    with os.scandir(path) as scandir_it:
         ^^^^^^^^^^^^^^^^
NotADirectoryError: [WinError 267] The directory name is invalid: 'C:\\Users\\kartben\\AppData\\Local\\Temp\\tmp96ltv01q\\Kconfig.modules'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\kartben\zephyrproject\.venv\Lib\site-packages\sphinx\cmd\build.py", line 280, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kartben\zephyrproject\.venv\Lib\site-packages\sphinx\application.py", line 268, in __init__
    self._init_builder()
  File "C:\Users\kartben\zephyrproject\.venv\Lib\site-packages\sphinx\application.py", line 341, in _init_builder
    self.events.emit('builder-inited')
  File "C:\Users\kartben\zephyrproject\.venv\Lib\site-packages\sphinx\events.py", line 107, in emit
    raise ExtensionError(__("Handler %r for event %r threw an exception") %
sphinx.errors.ExtensionError: Handler <function kconfig_build_resources at 0x000001C115A82F20> for event 'builder-inited' threw an exception (exception: [WinError 267] The directory name is invalid: 'C:\\Users\\kartben\\AppData\\Local\\Temp\\tmp96ltv01q\\Kconfig.modules')

Extension error (zephyr.kconfig):
Handler <function kconfig_build_resources at 0x000001C115A82F20> for event 'builder-inited' threw an exception (exception: [WinError 267] The directory name is invalid: 'C:\\Users\\kartben\\AppData\\Local\\Temp\\tmp96ltv01q\\Kconfig.modules')

FAILED: CMakeFiles/html C:/Users/kartben/zephyrproject/zephyr/doc/_build/CMakeFiles/html
cmd.exe /C "cd /D C:\Users\kartben\zephyrproject\zephyr\doc\_build && "C:\Program Files\CMake\bin\cmake.exe" -E env "DOXYGEN_EXECUTABLE=C:/Program Files/doxygen/bin/doxygen.exe" DOT_EXECUTABLE=C:/ProgramData/chocolatey/bin/dot.exe C:/Users/kartben/zephyrproject/.venv/Scripts/sphinx-build.exe -b html -c C:/Users/kartben/zephyrproject/zephyr/doc -d C:/Users/kartben/zephyrproject/zephyr/doc/_build/doctrees -w C:/Users/kartben/zephyrproject/zephyr/doc/_build/html.log -t development -j auto -W --keep-going -T C:/Users/kartben/zephyrproject/zephyr/doc/_build/src C:/Users/kartben/zephyrproject/zephyr/doc/_build/html"
ninja: build stopped: subcommand failed.
```

**Comment by kartben** [comment]
I am definitely tempted to say it's a bit of a PEBKAC too, as CI is not catching it, but OTH I can reproduce on three different workspaces / OSes. It looks like kconfiglib is getting confused. Again, `modules/Kconfig` doesn't sound quite right to me (why would it not need updating after this PR, since now it only includes `modules/cmsis/Kconfig` but not `cmsis_dsp` and `cmsis_nn`?)

**Comment by XenuIsWatching** [comment]
@kartben Please take a look at https://github.com/zephyrproject-rtos/zephyr/pull/63120

**Pull request merged**
